core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -7279,11 +7294,45 @@ public static boolean isAmbiguousException(Exception ex) {
       CallCopyingArgHandler argHandler =
           new CallCopyingArgHandler(call, false);
       call.getOperator().acceptCall(this, call, true, argHandler);
-      final SqlNode result = argHandler.result();
+      final SqlNode result = expandRow(argHandler.result());
```

Comment:
I find is a bit weird to have expandRow happen as a side-effect of calculatePermuteOffset; maybe this is a sign that the function name should be updated.